lib/portlayer/network/context.go

```diff
@@ -99,51 +119,52 @@ func isUnspecifiedSubnet(n *net.IPNet) bool {
 }
 
 func (c *Context) newScopeCommon(id, name, scopeType string, subnet *net.IPNet, gateway net.IP, dns []net.IP, ipam *IPAM, network object.NetworkReference) (*Scope, error) {
-	if isUnspecifiedSubnet(subnet) {
-		subnet = defaultSubnet
-	}
-
 	var err error
+	var space *AddressSpace
 
 	// allocate the subnet
-	space, defaultPool, err := c.reserveSubnet(subnet)
-	defer func() {
-		if err != nil && space != nil && defaultPool {
-			c.defaultBridgePool.ReleaseIP4Range(space)
+	if !isUnspecifiedSubnet(subnet) {
+		var defaultPool bool
+		space, defaultPool, err = c.reserveSubnet(subnet)
+		defer func() {
+			if err != nil && space != nil && defaultPool {
+				c.defaultBridgePool.ReleaseIP4Range(space)
+			}
+		}()
+
+		if err != nil {
+			return nil, err
 		}
-	}()
 
-	if err != nil {
-		return nil, err
-	}
+		subnet = space.Network
 
-	subnet = space.Network
+		// reserve the network and broadcast addresses
+		err = reserveBroadcastAndNetwork(space)
+		defer func() {
+			if err == nil || space.Network == nil {
+				return
+			}
 
-	// reserve the network and broadcast addresses
-	err = reserveBroadcastAndNetwork(space)
-	defer func() {
-		if err == nil || space.Network == nil {
-			return
+			lo := incrementIP4(space.Network.IP)
+			hi := decrementIP4(highestIP4(space.Network))
+			space.ReleaseIP4(lo)
+			space.ReleaseIP4(hi)
+		}()
+
+		if err != nil {
+			return nil, err
 		}
 
-		lo := incrementIP4(space.Network.IP)
-		hi := decrementIP4(highestIP4(space.Network))
-		space.ReleaseIP4(lo)
-		space.ReleaseIP4(hi)
-	}()
+		var subSpaces []*AddressSpace
+		subSpaces, err = reservePools(space, ipam)
+		if err != nil {
+			return nil, err
+		}
 
-	if err != nil {
-		return nil, err
+		ipam.spaces = subSpaces
 	}
 
-	subSpaces, err := reservePools(space, ipam)
-	if err != nil {
-		return nil, err
-	}
-
-	ipam.spaces = subSpaces
-
-	if gateway.IsUnspecified() {
+	if gateway.IsUnspecified() && len(ipam.spaces) > 0 {
```

Comment:
gateway.IsUnspecified will not work currently - the IP address unpacked from Config will be an empty array rather than {0, 0, 0, 0} - see https://play.golang.org/p/gU9_3_0y-n
I suggest for now that we write our own IsUnspecified as with IsUnspecifiedSubnet to check for nil, len(x) == 0, etc.